### ISSUE TITLE
Rule Change: no-unneeded-ternary for cases where optional chaining is the solution

### ISSUE THREAD

**Issue by legowerewolf** [eslint/eslint#19801]
### What rule do you want to change?

no-unneeded-ternary

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const link = field.links ? field.links['setAdvancedSearch'] : undefined;
```

### What does the rule currently do for this code?

Ignores it.

### What will the rule do after it's changed?

Warns about it, suggests that it be changed to

```js
const link = field.links?.['setAdvancedSearch'];
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the suggestion. The rule is behaving correctly based on its intended use, which is flag when ternary expressions are not needed. In this case, the ternary expression is needed even though it could also be written as optional chaining.

What we really need is to implement this rule that could be used in conjunction:
https://github.com/eslint/eslint/issues/13430
